test/gw/source_test.py

```diff
@@ -65,6 +71,21 @@ def test_waveform_error_raising(self):
                 self.frequency_array, **raise_error_parameters
             )
 
+    def test_unused_waveform_kwargs_message(self):
+        self.parameters.update(self.waveform_kwargs)
+        self.parameters["unused_waveform_parameter"] = 1.0
+        self._caplog.set_level(logging.WARNING, logger="bilby")
+        bilby.gw.source.logger.setLevel(logging.WARNING)
```

Comment:
I don't think these lines are necessary, since you set them below in the `with` statement.

```suggestion

```